mmcv/transforms/wrappers.py

```diff
@@ -0,0 +1,307 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+
+from collections.abc import Sequence
+from typing import Any, Callable, Dict, List, Optional, Union
+
+import numpy as np
+
+import mmcv
+from .base import BaseTransform
+from .builder import TRANSFORMS
+from .utils import cache_random_params
+
+# Indicator for required but missing keys in results
+NotInResults = object()
+
+# Import nullcontext if python>=3.7, otherwise use a simple alternative
+# implementation.
+try:
+    from contextlib import nullcontext
+except ImportError:
+    from contextlib import contextmanager
+
+    @contextmanager
+    def nullcontext(resource=None):
+        try:
+            yield resource
+        finally:
+            pass
+
+
+@TRANSFORMS.register_module()
+class Compose(BaseTransform):
+    """Compose a data pipeline with a sequence of transforms.
+
+    Args:
+        transforms (list[dict | callable]): Either config
+          dicts of transforms or transform objects.
+    """
+
+    def __init__(self, transforms: List[Union[Dict, Callable[[Dict], Dict]]]):
+        assert isinstance(transforms, Sequence)
+        self.transforms = []
+        for transform in transforms:
+            if isinstance(transform, dict):
+                transform = TRANSFORMS.build(transform)
+                self.transforms.append(transform)
+            elif callable(transform):
+                self.transforms.append(transform)
+            else:
+                raise TypeError('transform must be callable or a dict, but got'
+                                f' {type(transform)}')
+
+    def __iter__(self):
+        """Allow easy iteration over the transform sequence."""
+        return iter(self.transforms)
+
+    def transform(self, results):
+        """Call function to apply transforms sequentially.
+
+        Args:
+            results (dict): A result dict contains the results to transform.
+
+        Returns:
+            dict: Transformed results.
+        """
+        for t in self.transforms:
+            results = t(results)
+            if results is None:
+                return None
+        return results
+
+    def __repr__(self):
+        """Compute the string representation."""
+        format_string = self.__class__.__name__ + '('
+        for t in self.transforms:
+            format_string += f'\n    {t}'
+        format_string += '\n)'
+        return format_string
+
+
+@TRANSFORMS.register_module()
+class Remap(BaseTransform):
+    """A transform wrapper to remap and reorganize the input/output of the
+    wrapped transforms (or sub-pipeline).
+
+    Args:
+        transforms (list[dict|callable]):
+        input_mapping (dict): A dict that defines the input key mapping.
+            The keys corresponds to the inner key (i.e. kwargs of the
+            `transform` method), and the values corresponds to the outer
+            keys (i.e. the keys of the data/results).
```

Comment:
What are the types of values, strings or lists, or anything else?

Reply:
Values can be string, list, or dict. Will specify it in the docstring.